### PR TITLE
enable forwarding to different ports

### PR DESCRIPTION
**Pull request by makaleta**

### CODE REVIEW

**Comment by qoomon** [comment]
Hi @makaleta, that's a nice idea.

I would prefer to keep source and target port configuration within the `PORTS` variable. Something like `PORTS=80@8080,443@8443` or `PORTS=80>8080,443>8443`

File: entrypoint.sh [hunk 1]
```diff
@@ -68,17 +68,22 @@ then
...
   iptables -t nat -I POSTROUTING -j MASQUERADE
   for forwarding_port in ${FORWARDING_PORTS}
   do
+    host_port=`echo "$HOST_PORTS" | cut -d ' ' -f 1`
+    HOST_PORTS=`echo "$HOST_PORTS" | sed 's/[^ ]* *\(.*\)$/\1/'`
     iptables --table nat --insert PREROUTING \
       --protocol tcp \
       --dport "$forwarding_port" \
-      --jump DNAT --to-destination "$docker_host_ip"
+      --jump DNAT --to-destination "$docker_host_ip:$host_port"
```
Comment (qoomon) [inline entrypoint.sh:82]: for me it does not work for port ranges `8000:9000` even `8000-9000` will not work and ends in an error message ` Port `8000-9000' not valid`

**Comment by qoomon** [comment]
solution to support `PORTS=80@8080,443@8443`

```shell
for forwarding_port in ${FORWARDING_PORTS}
  do
    forwarding_ingress_port="$(echo "$forwarding_port" | cut -d'@' -f1)"
    forwarding_egress_port="$(echo "$forwarding_port" | cut -d'@' -f2)"
    forwarding_egress_port="${forwarding_egress_port:-$forwarding_ingress_port}"
    forwarding_egress_port="${forwarding_egress_port/:/-}"

    iptables --table nat --insert PREROUTING \
      --protocol tcp --destination-port "$forwarding_ingress_port" \
      --jump DNAT --to-destination "$docker_host_ip:$forwarding_egress_port"
    iptables --table nat --insert PREROUTING \
      --protocol udp --destination-port "$forwarding_ingress_port" \
      --jump DNAT --to-destination "$docker_host_ip:$forwarding_egress_port"
  done
```

File: entrypoint.sh (hunk 1 above)
Comment (qoomon) [reply entrypoint.sh:82]: it does not work for the default port range of `0:65535`

Comment (qoomon) [reply entrypoint.sh:82]: however it works for `1:65535`.  maybe i should change the default.  should port `0` be a valid value? WDYT?

**Comment by makaleta** [comment]
This alternative looks good. I will change PR accordingly.
Port 0 is reserved and has special use in POSIX, so probably shouldn't be in default value.

**Comment by qoomon** [comment]
LGTM, however I'm considering to change the syntax of `PORTS` values or the way to pass port mapping.

I'd like to align port mapping syntax (separators) to the docker parameter style of `--publish/-p` e.g.  `80:8080` (`INGRESS_PORT:EGRESS_PORT` )  and port ranges separator would be `-` e.g.  `8000-9000:5000-6000`

Furthermore I'd like to pass port mappings as container parameters instead of the `PORTS`environment variable.
```
docker run \
  --name 'docker-host' \
  --cap-add=NET_ADMIN --cap-add=NET_RAW \
  --restart on-failure \
  -d qoomon/docker-host \
  80:8080 8000-9000
```

WDYT?

**Pull request merged**